### PR TITLE
Harddel fixes

### PR DESCRIPTION
**Pull request by thgvr**
## About The Pull Request
Ports https://github.com/tgstation/tgstation/pull/56202
Ports https://github.com/tgstation/tgstation/pull/60067
Ports other stuff from Lemon on tgstation. He's a saint.
I tested this locally with some debugging and had good results.
Basically:

Fixes a harddel with pipenets
Fixes a harddel with canisters, and canisters in atmos machinery
Fixes a harddel with radial menus

## Changelog

:cl:
fix: Potentially fixed a lot of lag related to pipes
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/pipes/pipes.dm [hunk 1]
```diff
@@ -83,10 +83,10 @@
 	parent = P
 
 /obj/machinery/atmospherics/pipe/Destroy()
-	QDEL_NULL(parent)
+	parent = null
```
Comment (Kapu1178) [inline code/modules/atmospherics/machinery/pipes/pipes.dm]: isnt parent the pipenet? pretty sure this wants to stay

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/pipes/pipes.dm (hunk 1 above)
Comment (thgvr) [reply code/modules/atmospherics/machinery/pipes/pipes.dm:86]: You may be right

Comment (thgvr) [reply code/modules/atmospherics/machinery/pipes/pipes.dm]: You may be right

**Review by MarkSuckerberg — APPROVED** [review]
it's a start

**Pull request merged**